src/QuickMarcEditor/QuickMarcEditor.test.js

```diff
@@ -819,11 +826,13 @@ describe('Given QuickMarcEditor', () => {
 
           fireEvent.change(contentField, { target: { value: '1xx update' } });
           fireEvent.click(getByText('stripes-acq-components.FormFooter.save'));
-          fireEvent.click(getByText('Confirm'));
 
-          expect(onSubmitMock).toHaveBeenCalled();
-          waitFor(() => expect(queryByText('Confirmation modal')).toBeNull());
-          waitFor(() => expect(onCloseMock).toHaveBeenCalledWith());
+          waitFor(() => {
+            fireEvent.click(getByText('Confirm'));
+            expect(onSubmitMock).toHaveBeenCalled();
+            expect(queryByText('Confirmation modal'));
+            expect(onCloseMock).toHaveBeenCalledWith();
+          });
```

Comment:
`onCloseMock` is not invoked during this process, the test passes due to avoiding using `await` for `waitFor`. I was fixing these tests too in this [PR](https://github.com/folio-org/ui-quick-marc/pull/698/files).